src/site/content/en/blog/loading-wasm/index.md

```diff
@@ -0,0 +1,176 @@
+synchronous API, meaning it blocks the main thread until it completes. To discourage its use, Chrome
+disables `WebAssembly.Module` for buffers larger than 4 KB. To work around the size limit, we can
+use `await WebAssembly.compile(buffer)` instead:
+
+```js
+(async () => {
+  const response = await fetch('fibonacci.wasm');
+  const buffer = await response.arrayBuffer();
+  const module = await WebAssembly.compile(buffer);
+  const instance = new WebAssembly.Instance(module);
+  const result = instance.exports.fibonacci(42);
+  console.log(result);
+})();
+```
+
+`await WebAssembly.compile(buffer)` is _still_ not the optimal approach, but we’ll get to that in a
+second.
+
+Almost every operation in the modified snippet is now asynchronous, as the use of `await` makes
+clear. The only exception is `new WebAssembly.Instance(module)`, which has the same 4 KB buffer
+size restriction in Chrome. For consistency and for the sake of [keeping the main thread
+free](https://twitter.com/mathias/status/978549917332500480), we can use the asynchronous
+`WebAssembly.instantiate(module)`.
+
+```js
+(async () => {
+  const response = await fetch('fibonacci.wasm');
+  const buffer = await response.arrayBuffer();
+  const module = await WebAssembly.compile(buffer);
+  const instance = await WebAssembly.instantiate(module);
+  const result = instance.exports.fibonacci(42);
+  console.log(result);
+})();
+```
+
+Let’s get back to the `compile` optimization I hinted at earlier. With [streaming
+compilation](https://v8.dev/blog/v8-release-65), the browser can already
+start to compile the WebAssembly module while the module bytes are still downloading. Since download
+and compilation happen in parallel, this is faster — especially for large payloads.
+
+<figure>
+{% Img src="image/T4FyVKpzu4WKF1kBNvXepbi08t52/Y1FWFpD7bgeWaN95dXdL.png", alt="When the download time is
+longer than the compilation time of the WebAssembly module, then WebAssembly.compileStreaming()
+finishes compilation almost immediately after the last bytes are downloaded.", width="800", height="387" %}
+</figure>
+
+To enable this optimization, use `WebAssembly.compileStreaming` instead of `WebAssembly.compile`.
+This change also allows us to get rid of the intermediate array buffer, since we can now pass the
+`Response` instance returned by `await fetch(url)` directly.
+
+```js
+(async () => {
+  const response = await fetch('fibonacci.wasm');
+  const module = await WebAssembly.compileStreaming(response);
+  const instance = await WebAssembly.instantiate(module);
+  const result = instance.exports.fibonacci(42);
+  console.log(result);
+})();
+```
+
+Note: The server must be configured to serve the `.wasm` file with the correct MIME type by sending
+the `Content-Type: application/wasm` header. In previous examples, this wasn’t necessary since we
+were passing the response bytes as an array buffer, and so no MIME type checking took place.
```

Comment:
```suggestion
{% Aside %}
The server must be configured to serve the `.wasm` file with the correct MIME type by sending
the `Content-Type: application/wasm` header. In previous examples, this wasn’t necessary since we
were passing the response bytes as an array buffer, and so no MIME type checking took place.
{% endAside %}
```